viper/src/ast_factory/expression.rs

```diff
@@ -546,6 +555,25 @@ impl<'a> AstFactory<'a> {
         Expr::new(obj)
     }
 
+    pub fn backend_func_app(
+        &self,
+        backend_function_name: &str,
```

Comment:
@lukfried I think the crash happens because you should pass the actual backend function here (similarly to `domain_func_app), not just its name. See the signature of [Scala constructor](https://github.com/viperproject/silver/pull/428/files#diff-20523c8dac890ebb6c98a54fbabb02db236d0863632d3cc112c5a837f547ac0fR390).